### PR TITLE
[NO GBP]Fixes plumbing for good(hopefully) & more reagent code

### LINKED ISSUE #78945: Chemical factories still appear to be broken
## Round ID: N/A

Chemical factories still appear to be broken, even after the PR that was supposed to fix them merged.

4Ihs7C6
This reaction chamber is supposed to fill to that amount provided, but seems to just stop right before it's supposed to with the water and locks up, and even overfills with the hydrogen.

KIbtQWI
This chemical press is set to 50 units, but then stops at 49.9998 units and locks up.

These are just some examples but it happens to quite a few things with factories.

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
1. This should be the end of all problems related to plumbing and I swear if another issue popes up I'm goanna delete my GitHub account. Or to be more realistic just ping me and I'll be back :D

   - Fixes #78945
  
    And this time I am **100%**(if I'm wrong then fuck me) sure that reagents will not flow in excess amounts into reaction chambers/synthesizers or whatever plumbing component you can think off.
    
    Reagents flowing in excess amounts is what caused the factory to grind to a halt as the components will stop processing if they detect excess amounts of reagents (which is why plunging them with a plunger would clear them of their reagents and help them resume processing again which obviously is not an ideal solution)
    
    Also, now it's no longer required to see reagent volumes in 4 decimal places so I rounded it back to just 2 places again so values like 0.999 or 1.001 will become just 1 again. 
    
    Also plumbing reaction chamber was doing too much work so I fixed that.

2. Made defines for min & max ph. any change to ph values of any reagent will be confined between these values i.e., 0 & 14. Also converted some vars into defines. Using a variable makes sense if it's value changes but if it doesn't then just define it as a constant so you can save memory. Made synthesizer lists static to further save memory.

3. Significantly boosted the ph balancing mechanism for reaction chamber. You will be surprised. 

## Changelog
:cl:
fix: plumbing setups should(hopefully) no longer grind to a halt nor will overflow with excess volume of reagents.
code: created defines for min & max ph. Improved some reaction_reagent code. Made synthesizer dispensable reagent list values static to save memory.
refactor: ph balancing mechanism for reaction chamber is significantly improved. Optimized it's code overall
refactor: examining each individual reagent will display their results back to 2 decimal places again and not 4 for easy readability.
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
@Time-Green since plumbing is your speciality i need your advice on some things

https://github.com/tgstation/tgstation/blob/278ab845915eef463617e0351f46dcd3bf2c2985/code/modules/plumbing/plumbers/synthesizer.dm#L54

Synthesizer's assume that `seconds_per_tick` will always be `2` which is why they halved it to bring it back to 1. This assumption seems dangerous if `seconds_per_tick` changes during the game so my question is
- should i remove usage of  `seconds_per_tick` completly?
- Is there some other newer math were we can still use `seconds_per_tick` to get the correct results

**Comment by MidoriWroth** [comment]
Thank you for your efforts!

**Comment by SyncIt21** [comment]
Well time green schedule is pretty packed so i won't bother him with this PR. I'll undraft it

**Comment by SyncIt21** [comment]
@Time-Green got the time?

File: code/modules/reagents/chemistry/reagents/reaction_agents_reagents.dm [hunk 1]
```diff
@@ -24,22 +24,23 @@
...
-		target.my_atom.audible_message(span_warning("The beaker froths as the buffer is added, to no effect."))
-		playsound(target.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
-		holder.remove_reagent(type, amount)//Remove from holder because it's not transferred
-		return
-	var/ph_change = -((amount/target.total_volume)*strength)
-	target.adjust_all_reagents_ph(ph_change, ph, 14)
-	target.my_atom.audible_message(span_warning("The beaker fizzes as the ph changes!"))
+		message = "The beaker froths as the buffer is added, to no effect."
+	else
+		message = "The beaker froths as the ph changes!"
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents/reaction_agents_reagents.dm:39]:
```suggestion
		message = "The beaker froths as the pH changes!"
```

File: code/modules/reagents/chemistry/reagents/reaction_agents_reagents.dm [hunk 2]
```diff
@@ -51,21 +52,22 @@
...
-		target.my_atom.audible_message(span_warning("The beaker froths as the buffer is added, to no effect."))
-		playsound(target.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
-		holder.remove_reagent(type, amount)//Remove from holder because it's not transferred
-		return
-	var/ph_change = (amount/target.total_volume)*strength
-	target.adjust_all_reagents_ph(ph_change, 0, ph)
-	target.my_atom.audible_message(span_warning("The beaker froths as the ph changes!"))
+		message = "The beaker froths as the buffer is added, to no effect."
+	else
+		message = "The beaker froths as the ph changes!"
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents/reaction_agents_reagents.dm:66]:
```suggestion
		message = "The beaker froths as the pH changes!"
```

**Review by Ghommie — APPROVED** [review]

**Comment by Time-Green** [comment]
Sorry, I’ve been very distracted. Ghommie is the bomb though

**Pull request merged**